tests/integration/golang/mlflow/artifact/get_artifact_s3_test.go

```diff
@@ -0,0 +1,194 @@
+//go:build integration
+
+package artifact
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/google/uuid"
+	"github.com/hetiansu5/urlquery"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api/request"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/fixtures"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/helpers"
+)
+
+type GetArtifactS3TestSuite struct {
+	suite.Suite
+	s3Client           *s3.Client
+	runFixtures        *fixtures.RunFixtures
+	serviceClient      *helpers.HttpClient
+	experimentFixtures *fixtures.ExperimentFixtures
+}
+
+func TestGetArtifactTestSuite(t *testing.T) {
+	suite.Run(t, new(GetArtifactS3TestSuite))
+}
+
+func (s *GetArtifactS3TestSuite) SetupTest() {
+	s3Client, err := helpers.NewS3Client(helpers.GetS3EndpointUri())
+	assert.Nil(s.T(), err)
+	s.s3Client = s3Client
+
+	s.serviceClient = helpers.NewMlflowApiClient(helpers.GetServiceUri())
+
+	experimentFixtures, err := fixtures.NewExperimentFixtures(helpers.GetDatabaseUri())
+	assert.Nil(s.T(), err)
+	s.experimentFixtures = experimentFixtures
+
+	runFixtures, err := fixtures.NewRunFixtures(helpers.GetDatabaseUri())
+	assert.Nil(s.T(), err)
+	s.runFixtures = runFixtures
+}
+
+func (s *GetArtifactS3TestSuite) Test_Ok() {
+	defer func() {
+		assert.Nil(s.T(), s.experimentFixtures.UnloadFixtures())
+	}()
+
+	testData := []struct {
+		name   string
+		bucket string
+	}{
+		{
+			name:   "TestWithBucket1",
+			bucket: "bucket1",
+		},
+		{
+			name:   "TestWithBucket2",
+			bucket: "bucket2",
+		},
+	}
+
+	for _, tt := range testData {
```

Comment:
and the same question probably here? we can setup `experiment`, `run` and all other data only once? not for each test case?